widgetastic_manageiq/__init__.py

```diff
@@ -2820,3 +2816,83 @@ def fill(self, value):
             return False
         self.select_button(value)
         return True
+
+
+class BaseNonInteractiveEntitiesView(View, ReportDataControllerMixin):
+    """Represents Quadicons appearing in views like Edit Tags, etc.
+
+    """
+    elements = ('//tr[./td/div[@class="quadicon"]]/following-sibling::tr/td/*[self::a or '
+                'self::span]')
+
+    @property
+    def entity_class(self):
+        if self.browser.product_version < '5.9':
+            return NonJSBaseEntity
+        else:
+            return JSBaseEntity
+
+    @property
+    def entity_names(self):
+        """ looks for entities and extracts their names
+
+        Returns: all current page entities
+        """
+        if self.browser.product_version < '5.9':
+            br = self.browser
+            return [br.get_attribute('title', el) for el in br.elements(self.elements)]
+        else:
+            entities = self._invoke_cmd('get_all_items')
+            return [entity['item']['cells']['Name'] for entity in entities]
+
+    def get_all(self):
+        """ obtains all entities like QuadIcon displayed by view
+
+        Returns: all entities (QuadIcon/etc.) displayed by view
+        """
+        return [self.entity_class(parent=self, name=name) for name in self.entity_names]
+
+    def get_entities(self, by_name=None):
+        """ obtains all matched entities like QuadIcon displayed by view
+        Args:
+            by_name (str): only entities which match to by_name will be returned
+
+        Returns: all matched entities (QuadIcon/etc.) displayed by view
+        """
+        entities = self.get_all()
+        remaining_entities = []
+        for entity in entities:
+            if by_name and by_name in entity.name:
+                remaining_entities.append(entity)
+                # todo: by_type and by_regexp will be implemented later if needed
+        return remaining_entities
+
+    def get_entity(self, by_name=None):
+        """ obtains one entity matched to by_name
+        raises exception if no entities or several entities were found
+        Args:
+            by_name (str): only entity which match to by_name will be returned
+
+        Returns: matched entities (QuadIcon/etc.)
+        """
+        entities = self.get_entities(by_name=by_name)
+        if len(entities) == 0:
```

Comment:
Consider moving this into `get_entities` so its also raised by that method when called directly and nothing matches.